src/serde_helpers.rs

```diff
@@ -4,6 +4,11 @@ use std::{convert::TryFrom, result::Result};
 
 use serde::{ser, Serialize, Serializer};
 
+#[cfg(feature = "uuid-0_8")]
```

Comment:
I thought we'd decided in discussion that it was okay for these to be mutually exclusive; that's certainly what was recorded in RUST-1305.  Is that not accurate?

I don't see how the dependency union is an issue here; the user of this library will have a single version of `uuid` that they're working with and they can select that as the one the `bson` interface accepts; if other dependencies happen to use a different version internally that won't matter.

Reply:
I don't recall that but it's possible there was a miscommunication. The reason it's an issue is that Cargo treats `0.x` and `1.x` as essentially different crates, so libraries can depend on either or both versions of `uuid`. This means you could have a situation where dependency A (either direct or transitive) depends on `bson` with `uuid-1` flag enabled and dependency B (again direct or transitive) depends on `bson` with `uuid-0_8` flag on, and this would fail to compile due to the current mutually exclusive implementation. Since this covers the entire dependency graph and is basically out of an individual user's control, I think it's important that both features are purely additive in nature. The rust book has some [good info](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification) on this problem.

Reply:
Blarg, you're right, I hadn't considered the case where `bson` was the crate pulled in multiple times.  Updated - reduced some of the repetition with macros but there's still a _lot_ of boilerplate now.  We should consider dropping uuid 0.8 support for bson 3.0.

Reply:
Yep, definitely. I filed RUST-1380 to track that.